host/class/cdc/usb_host_cdc_acm/host_test/device_interaction/main/test_opening_device.cpp

```diff
@@ -0,0 +1,123 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
+#include <catch2/catch_test_macros.hpp>
+
+#include "cdc_descriptors.hpp"
+#include "usb/cdc_acm_host.h"
+#include "mock_add_usb_device.h"
+#include "common_test_fixtures.hpp"
+
+extern "C" {
+#include "Mockusb_host.h"
+}
+
+/**
+ * @brief Add mocked devices
+ *
+ * Mocked devices are defined by a device descriptor, a configuration descriptor and a device address
+ */
+static void _add_mocked_devices(void)
+{
+    // Init mocked devices list at the beginning of the test
+    usb_host_mock_dev_list_init();
+
+    // Fill mocked devices list
+    // CP210x (FS descriptor)
+    REQUIRE(ESP_OK == usb_host_mock_add_device(5, (const usb_device_desc_t *)cp210x_device_desc,
+            (const usb_config_desc_t *)cp210x_config_desc));
+}
+
+
+SCENARIO("Test mocked device opening and closing")
```

Comment:
Can we add at least 1 test where the device is successfully opened?

`REQUIRE(ESP_OK == cdc_acm_host_open(.....`

Reply:
I did not add `cdc_acm_host_open` in purpose, because we are opening and closing devices in test cases present in `test_device_interactions.cpp`, so it would be mostly copy-paste of `test_cdc_acm_host_open` fuction.

But, anyway... we can have it here as well, no problem